### ISSUE TITLE
eval log

### ISSUE THREAD

**Issue by zgsxwsdxg** [PaddlePaddle/PaddleSeg#2511]
paddleseg/core/train.py 文件
if mean_iou > best_mean_iou:
                        best_mean_iou = mean_iou
                        best_model_iter = iter
                        best_model_dir = os.path.join(save_dir, "best_model")
                        paddle.save(
                            model.state_dict(),
                            os.path.join(best_model_dir, 'model.pdparams'))
                    logger.info(
                        '[EVAL] The model with the best validation mIoU ({:.6f}) was saved at iter {}.'
                        .format(best_mean_iou, best_model_iter))

我有点疑惑，为什么我训练过程中总会打印'[EVAL] The model with the best validation mIoU ({:.6f}) was saved at iter {}. 这一句，但是iter始终是1000，mIOU也一直不变。我感觉是不是那块逻辑有问题，既然进入mean_iou > best_mean_iou条件了，那么best_mean_iou, best_model_iter 应该是变化的吧？你们看看是不是在保存最好模型这块有点问题？谢谢

**Comment by zgsxwsdxg** [comment]
> paddleseg/core/train.py 文件 if mean_iou > best_mean_iou: best_mean_iou = mean_iou best_model_iter = iter best_model_dir = os.path.join(save_dir, "best_model") paddle.save( model.state_dict(), os.path.join(best_model_dir, 'model.pdparams')) logger.info( '[EVAL] The model with the best validation mIoU ({:.6f}) was saved at iter {}.' .format(best_mean_iou, best_model_iter))
> 
> 我有点疑惑，为什么我训练过程中总会打印'[EVAL] The model with the best validation mIoU ({:.6f}) was saved at iter {}. 这一句，但是iter始终是1000，mIOU也一直不变。我感觉是不是那块逻辑有问题，既然进入mean_iou > best_mean_iou条件了，那么best_mean_iou, best_model_iter 应该是变化的吧？你们看看是不是在保存最好模型这块有点问题？谢谢

打印的内容为 [EVAL] The model with the best validation mIoU (0.5000) was saved at iter 1000

**Comment by haoyuying** [comment]
是不是你本身的训练结果确实是1000iter最好，这里的代码我们已经用过多次，是没有问题的

**Comment by zgsxwsdxg** [comment]
> 是不是你本身的训练结果确实是1000iter最好，这里的代码我们已经用过多次，是没有问题的

你好像没看逻辑，是这样的，就算我是1000iter最好，那么在大于1000后怎么会进入if mean_iou > best_mean_iou: 这个条件，说明1000后的模型mean_iou> best_mean_iou，并且代码best_mean_iou = mean_iou best_model_iter = iter是更新最新的iou和iter，但是log输出始终是[EVAL] The model with the best validation mIoU (0.5000) was saved at iter 1000，不知你明白我说的啥不？谢谢，细心看看这块的代码逻辑，

**Comment by wuyefeilin** [comment]
感谢您的反馈，经过再三确认，代码没有问题。每次eval，都会打印出历史最佳模型的mIoU及其出现的所在iter。

**Comment by zgsxwsdxg** [comment]
> 感谢您的反馈，经过再三确认，代码没有问题。每次eval，都会打印出历史最佳模型的mIoU及其出现的所在iter。

谢谢您的回答，没有问题就好，顺便问一下：在配置训练文件中
transforms:
    - type: RandomHorizontalFlip
    - type: RandomVerticalFlip
    - type: RandomDistort
      brightness_range: 0.4
      contrast_range: 0.4
      saturation_range: 0.4
     - type: RandomRotation
       max_rotation: 10
       im_padding_value: (0,0,0)
       label_padding_value: 0
    - type: RandomBlur
    - type: RandomScaleAspect
      min_scale: 0.8
      aspect_ratio: 0.5

如果放开RandomRotation 训练会报错，如果注释后训练没问题，这个怎么解决？谢谢

**Comment by wuyefeilin** [comment]
rotation后会改变图片大小，你可以是用crop或者resize等操作，让一个batch的图片大小一致

**Comment by zgsxwsdxg** [comment]
> rotation后会改变图片大小，你可以是用crop或者resize等操作，让一个batch的图片大小一致

没太明白，我的transform是这样的：
 transforms:
    - type: RandomHorizontalFlip
    - type: RandomVerticalFlip
    - type: RandomDistort
      brightness_range: 0.4
      contrast_range: 0.4
      saturation_range: 0.4
    - type: RandomRotation
      max_rotation: 10
      im_padding_value: (0,0,0)
      label_padding_value: 0
    - type: RandomBlur
    - type: RandomScaleAspect
      min_scale: 0.8
      aspect_ratio: 0.5

    - type: Resize
      target_size: [ 512, 512 ]
    - type: Normalize
改哪个？改为啥？谢谢

**Comment by zgsxwsdxg** [comment]
> > rotation后会改变图片大小，你可以是用crop或者resize等操作，让一个batch的图片大小一致
> 
> 没太明白，我的transform是这样的： transforms: - type: RandomHorizontalFlip - type: RandomVerticalFlip - type: RandomDistort brightness_range: 0.4 contrast_range: 0.4 saturation_range: 0.4 - type: RandomRotation max_rotation: 10 im_padding_value: (0,0,0) label_padding_value: 0 - type: RandomBlur - type: RandomScaleAspect min_scale: 0.8 aspect_ratio: 0.5
> 
> ```
> - type: Resize
>   target_size: [ 512, 512 ]
> - type: Normalize
> ```
> 
> 改哪个？改为啥？谢谢



> rotation后会改变图片大小，你可以是用crop或者resize等操作，让一个batch的图片大小一致

其实我后面是有resize的，不知如何改？

**Comment by zgsxwsdxg** [comment]
> rotation后会改变图片大小，你可以是用crop或者resize等操作，让一个batch的图片大小一致

我yml配置文件是这个样子：
` transforms:
    - type: RandomHorizontalFlip
    - type: RandomVerticalFlip
    - type: RandomDistort
      brightness_range: 0.4
      contrast_range: 0.4
      saturation_range: 0.4
    - type: RandomRotation
      max_rotation: 10
      im_padding_value: (0,0,0)
      label_padding_value: 0
    - type: RandomBlur
    - type: RandomScaleAspect
      min_scale: 0.8
      aspect_ratio: 0.5
    - type: Resize
      target_size: [ 512, 512 ]
    - type: Normalize
    `

**Comment by wuyefeilin** [comment]
请给一下具体的报错信息

**Comment by zgsxwsdxg** [comment]
> 请给一下具体的报错信息

W0907 15:03:09.787272 1387109 gpu_resources.cc:61] Please NOTE: device: 0, GPU Compute Capability: 7.0, Driver API Version: 11.2, Runtime API Version: 11.2
W0907 15:03:09.787304 1387109 gpu_resources.cc:91] device: 0, cuDNN Version: 8.1.
2022-09-07 15:03:11 [INFO]      Loading pretrained model from https://bj.bcebos.com/paddleseg/dygraph/hrnet_w18_ssld.tar.gz
2022-09-07 15:03:12 [INFO]      There are 1525/1525 variables loaded into HRNet.

    self.run()
  File "/data/duxiaogang/anaconda3/envs/ppseg/lib/python3.7/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/data/duxiaogang/anaconda3/envs/ppseg/lib/python3.7/site-packages/paddle/fluid/dataloader/dataloader_iter.py", line 218, in _thread_loop
    self._thread_done_event)
  File "/data/duxiaogang/anaconda3/envs/ppseg/lib/python3.7/site-packages/paddle/fluid/dataloader/fetcher.py", line 121, in fetch
    data.append(self.dataset[idx])
  File "/data/duxiaogang/PaddleSeg/paddleseg/datasets/dataset.py", line 155, in __getitem__
    data = self.transforms(data)
  File "/data/duxiaogang/PaddleSeg/paddleseg/transforms/transforms.py", line 78, in __call__
    data = op(data)
  File "/data/duxiaogang/PaddleSeg/paddleseg/transforms/transforms.py", line 838, in __call__
    borderValue=self.im_padding_value)
cv2.error: OpenCV(4.6.0) :-1: error: (-5:Bad argument) in function 'warpAffine'
> Overload resolution failed:
>  - Scalar value for argument 'borderValue' is longer than 4
>  - Scalar value for argument 'borderValue' is longer than 4
这是报错信息，帮忙分析下，谢谢您

**Comment by zgsxwsdxg** [comment]
> 请给一下具体的报错信息

自己解决了，不过还是要感谢你的耐心解答
